client/state/billing-data/reducer.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	BILLING_DATA_RECEIVE,
+	BILLING_DATA_REQUEST,
+	BILLING_DATA_REQUEST_FAILURE,
+	BILLING_DATA_REQUEST_SUCCESS
+} from 'state/action-types';
+import { billingDataItemsSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+/**
+ * Returns the updated items state after an action has been dispatched.
+ * The state contains all past and upcoming billing data entries.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const items = createReducer( {}, {
+	[ BILLING_DATA_RECEIVE ]: ( state, { past, upcoming } ) => ( { ...state, past, upcoming } ),
```

Comment:
Do we store any state besides past and upcoming? If not this can be simplified to:
```
{ past, upcoming };
```